extensions-core/multi-stage-query/src/main/java/org/apache/druid/msq/exec/Worker.java

```diff
@@ -67,6 +69,17 @@
    */
   void postWorkOrder(WorkOrder workOrder);
 
+  /**
+   * Returns the statistics snapshot for the given stageId
+   */
+  ClusterByStatisticsSnapshot fetchStatisticsSnapshot(StageId stageId) throws ExecutionException, InterruptedException;
+
+  /**
+   * Returns the statistics snapshot for the given stageId which contains only the sketch for the specified timeChunk
```

Comment:
Please also document which mode is this API getting used in.

Reply:
Done